### PR TITLE
hwdef: enable ROMFS if we have a defaults filepath to embed

### PR DESCRIPTION
**Pull request by peterbarker**
Corrects defaults on MatekL431-RC

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py [hunk 1]
```diff
@@ -2591,6 +2591,7 @@ def write_hwdef_header(self, outfilename):
 
         if os.path.exists(self.processed_defaults_filepath()):
             self.write_define(f, 'AP_PARAM_DEFAULTS_FILE_PARSING_ENABLED', 1)
+            self.write_define(f, 'AP_FILESYSTEM_ROMFS_ENABLED', 1)
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py:2594]: as we discussed, we should be using the direct AP_ROMFS API

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_Param/AP_Param.cpp [hunk 2]
```diff
@@ -1577,6 +1575,44 @@ void AP_Param::reload_defaults_file(bool last_pass)
...
+    // filename without the prefix:
+    const char *trimmed_filename = &default_file[strlen(prefix)];
+
+    uint32_t string_length;
+    const uint8_t *text = AP_ROMFS::find_decompress(trimmed_filename, string_length);
+    if (text == nullptr) {
+        return;
+    }
+
+    load_param_defaults((const char*)text, string_length, last_pass);
```
Comment (tridge) [inline libraries/AP_Param/AP_Param.cpp:1613]: missing AP_ROMFS::free() in case it is decompressed to allocated memory

**Review by tridge — CHANGES_REQUESTED** [review]
looks good apart from the missing free call, thanks!

**Comment by tridge** [comment]
I tested this on C-RTK2-HP, current master doesn't honor defaults.parm, this PR does

File: libraries/AP_Param/AP_Param.cpp (hunk 2 above)
Comment (peterbarker) [reply libraries/AP_Param/AP_Param.cpp:1613]: Fixed

**Comment by peterbarker** [comment]
L431 was tested with/without defaults
Pixhawk6X was tested with/without defaults

**Review by tridge — APPROVED** [review]

**Pull request merged**